### ISSUE TITLE
QRCode margin and text case

### ISSUE THREAD

**Issue by jfvoliveira** [SimpleSoftwareIO/simple-qrcode#85]
Hi there. I am having an issue with the size of the QRCode. Some codes are of the size I gave them with zero margin, others have a margin (making the code smaller).

Example:

```
<img src="data:image/png;base64,{!! base64_encode(QrCode::format('png')->errorCorrection('Q')->size('100')->margin(0)->generate('DEMO-4444-55555-333-AAA-AAA')) !!}">
<img src="data:image/png;base64,{!! base64_encode(QrCode::format('png')->errorCorrection('Q')->size('100')->margin(0)->generate('DEMO-4444-55555-333-BBB-AAa')) !!}">
<img src="data:image/png;base64,{!! base64_encode(QrCode::format('png')->errorCorrection('Q')->size('100')->margin(0)->generate('DEMO-4444-55555-333-CCC-AaA')) !!}">
<img src="data:image/png;base64,{!! base64_encode(QrCode::format('png')->errorCorrection('Q')->size('100')->margin(0)->generate('DEMO-4444-55555-333-DDD-aAA')) !!}">
<img src="data:image/png;base64,{!! base64_encode(QrCode::format('png')->errorCorrection('Q')->size('100')->margin(0)->generate('DEMO-4444-55555-333-EEE-aaa')) !!}">
<img src="data:image/png;base64,{!! base64_encode(QrCode::format('png')->errorCorrection('Q')->size('100')->margin(0)->generate('DEMO-4444-55555-333-EEE-BBB')) !!}">
```
The result is:

![Result](https://i.imgur.com/JArjzVW.png)

Why is there a margin in the codes that have lowercase? I've run a few tests and everytime a smallcase letter appears, the code gets a margin.
If I for instance remove the "DEMO-4444-5555-" from the start of the text, every code keeps the same size.

**Comment by SimplyCorey** [comment]
That's strange.  I'll take a look at it and see if there is a bug with the margin method.

**Comment by jfvoliveira** [comment]
Any news on this subject?

**Comment by SimplyCorey** [comment]
I have not had time to look into it.  Feel free to open a PR if you are able to solve the issue.

**Comment by chonchol** [comment]
I have generated 5k qrcode show it with dynamically below way. But if i open qrcodes page it takes huge time to open it. Because all qrcode make that time. How can i do that? Will i save qrcode into database then retrieve it? How can i save qrcode into databse?
`<img src="data:image/png;base64, {!! base64_encode(QrCode::format('png')->size(100)->generate($redirectedPage.$qrcode->id)) !!} ">`

**Comment by SimplyCorey** [comment]
@chonchol You can provide a second argument to the generate to save the file.  `generate('redirect code, 'filename'); You can just then serve these as normal files.

**Comment by SimplyCorey** [comment]
There is a release coming from upstream that should address his shortly.

**Comment by stevevg** [comment]
I'm experiencing the same issue as @jfvoliveira, but no time to look into it and try to solve it.

> There is a release coming from upstream that should address his shortly.

Was this referred to the original issue or to the one opened by @chonchol? Because I've just tried with v2.0.0 and the issue is still here.

**Comment by SimplyCorey** [comment]
I am waiting on a PR up stream from a different project.

**Comment by emilas44** [comment]
I also have a problem with margin... I made a small app with laravel that generates QrCodes from a .csv file.
There is a loop that goes through each row and all generated images have proper margin except one.
And what is most interesting, when I take that row from .csv and make a new .csv file it works fine. It makes the margin as I set it to.

I tried also playing with encoding and then it happens that there are more images created without applying the margin (they get some default margin).

**Comment by emilas44** [comment]
OK so I finally figured out why the margins are sometimes not applied... It's because of the amount of data that is crambled inside the string that needs to be generated. If there is too much text then the margins somehow don't get applied. So much from me!

**Comment by DASPRiD** [comment]
As commented on the upstream issue, this should be fixed in version 2.0.0 of BaconQrCode.

**Comment by SimplyCorey** [comment]
@DASPRiD Thanks, I plan on upgrading this week!

**Comment by pietrobp** [comment]
can u remove the border from qrcode? i put margin(0) but it continues with some white border/padding

**Comment by SimplyThomas** [comment]
This will be addressed in the 3.0.0 milestone

**Comment by SimplyCorey** [comment]
This is resolved in v3 which I am finishing up the docs for and will push up this week.
